likecoin/js/admin/likecoin_metabox.js

```diff
@@ -76,15 +128,57 @@ function onSubmitToISCN(e) {
   window.addEventListener('message', onISCNCallback, false);
 }
 
+async function onEstimateAndUploadArweave(e) {
+  e.preventDefault();
+  try {
+    const res = await jQuery.ajax({
+      type: 'POST',
+      url: `${wpApiSettings.root}likecoin/v1/posts/${wpApiSettings.postId}/arweave/estimate`,
+      method: 'POST',
+      beforeSend: (xhr) => {
+        xhr.setRequestHeader('X-WP-Nonce', wpApiSettings.nonce);
+      },
+    });
+    const { ipfsHash, LIKE, memo } = res;
+    if (LIKE === '0' || LIKE === 0) {
+      // exact same files are already exist in Arweave.
+      // skip Arweave upload flow to directly register an ISCN
+      const { arweaveId } = res;
+      const arweaveInfo = {
+        ipfsHash,
+        arweaveId,
+      };
+      submitToISCNAfterArweaveUpload(arweaveInfo);
+      return;
+    }
+    const { siteurl } = wpApiSettings;
+    const redirectString = encodeURIComponent(siteurl);
+    const likePayWidget = `http://localhost:3000/in/widget/pay?to=like-arweave&amount=${LIKE}&remarks=${memo}&opener=1&redirect_uri=${redirectString}`; // TODO: Change to test and mainnet.
```

Comment:
Please change to production url

Reply:
I think The `memo` needs to be `encodeURIComponent()` to escape special chars in memo